students/kegan/session03/slicing_lab.py

```diff
@@ -0,0 +1,154 @@
+"""
+Kathryn Egan
+
+Write some functions that take a sequence as an
+argument, and return a copy of that sequence:
+
+- with the first and last items exchanged
+- with every other item removed
+- with the first and last 4 items removed, and every
+other item in between
+- with the elements reversed (just with slicing)
+- with the middle third, then last third, then the
+first third in the new order
+"""
+
+
+def exchange(items):
+    """ Returns list with first and last positions
+    transposed.
+    Args:
+        items (list) : list to alter
+    Returns:
+        list : items with first and last value transposed
+    """
+    if len(items) < 2:
+        return items
+    return items[-1:] + items[1:-1] + items[:1]
+
+
+def remove_every_other(items):
+    """ Returns a list with every other item removed.
+    Args:
+        items (list) : list of items to alter
+    Returns:
+        list : return items with every other value removed
+    """
+    return items[::2]
+
+
+def first_last_removed(items):
+    """ Returns a new list with the first and last
+    four items removed and the rest with every other
+    value removed. Returns list with every other
+    element removed if length is <= 8.
+    Args:
+        items (list) : list of items to alter
+    Returns:
+        list : altered items
+    """
+    if len(items) <= 8:
+        return items[::2]
+    return items[4:-4:2]
+
+
+def reverse_it(items):
+    """ Returns a reversed version of the given items.
+    Args:
+        items (list) : list of items to reverse
+    Returns:
+        list : reversed items
+    """
+    return items[::-1]
+
+
+def thirds(items):
+    """ Returns a new list that is the middle and last
+    thirds of the given items then the non-greedy
+    first third. Original list if len is < 3.
+    Args:
+        items (list) : list of items
+    Returns:
+        list : middle and last followed by first third
+    """
+    if len(items) < 3:
+        return items
+    splice = len(items) // 3
+    first = items[:splice]
+    rest = items[splice:]
+    return rest + first
+
+
+def tuplify(items):
```

Comment:
both the tuple() and list() constructors take an arbitrary sequence --no need to write a loop to build it up:

```
In [20]: tuple("this is a sequence of chars")
Out[20]: 
('t',
 'h',
 'i',
 's',
 ' ',
 'i',
 's',
...
```